### PR TITLE
Disconnecting a Stripe account no longer revokes GiveWP as an Authorized Application

### LINKED ISSUE #5198: Disconnecting a single site from Stripe should not require disconnecting all sites on the same account
## User Story

As an admin of several sites, I want the ability to disconnect a single site from Stripe without having to disconnect all of my previously connected sites that are connected to my same Stripe account.

## Details

Currently, disconnecting a connected Stripe account from the GiveWP plugin revokes permissions for GiveWP as an "Authorized applications" on that Stripe account. This is destructive behaviour that breaks other connections to GiveWP on that account.

## Expected Behavior

Disconnecting a connect Stripe account from the GiveWP plugin should not revoke permissions for GiveWP as an "Authorized applications" on that Stripe account so that other connected sites continue to operate as expected.

## Steps to Reproduce

1. Go to Settings -> Payment Gateways -> Stripe -> Manage Accounts.
1. Click "Connect with Stripe" and complete the connect process.
1. Go to Settings -> [Authorized Applications](https://dashboard.stripe.com/account/applications) in the Stripe dashboard of the connected account.
1. View GiveWP as an Authorized Application.
1. Go to Settings -> Payment Gateways -> Stripe -> Manage Accounts.
1. Click "Disconnect" to remove a connected account.
1. Go to Settings -> [Authorized Applications](https://dashboard.stripe.com/account/applications) in the Stripe dashboard of the connected account.
1. View GiveWP is no longer an Authorized Application.

## Visuals

GiveWP connected Stripe accounts
image

Stripe authorized applications
image

## Additional Context

The Stripe add-on includes the option to connect a site via key.

## Acceptance Criteria

-   [x] A Stripe account can be disconnected form the site.
-   [x] Disconnecting an account does not revoke GiveWP as an Authorized Application.
-   [ ] Manually connected Stripe accounts (Stripe add-on) can still be disconnected.

### PR DESCRIPTION
**Pull request by kjohnson**
Resolves #5198

## Description

This change allows for Stripe accounts to be disconnected form a site while preserving GiveWP as an Authorized Application to prevent other connected sites from breaking.

Additionally, this change offers instructions for revoking GiveWP as an Authorized Application in an explicit manner.

**Note**: This does NOT change the behaviour of which Stripe accounts are listed with a Disconnect action. The behaviour is the same as `develop`.

- Disconnects a Stripe account using `give_stripe_disconnect_account`, which does not revoke GiveWP as an Authorized Application in Stripe.
- Adds a new `give_disconnect_connected_stripe_account` action with corresponding method callback.
- Adds a notice below the Stripe account list with instructions on revoking GiveWP as an Authorized Application.

Note: This is a fork of #5201. The big difference is that there is no longer an option to "Remove All" which originally was intended to Revoke access but only worked for the default connected account. Instead, the notice now serves as discover-able, inline documentation for users that want to revoke GiveWP as an Authorized Application.

## Affects

Disconnecting a site no longer revokes GiveWP as an Authorized Application.

## Visuals

| Stripe Settings |
| --- |
| **Multiple connected Stripe accounts** Screenshot_2020-10-14 GiveWP Settings ‹ WordPress — WordPress(2) |
| **A single connected Stripe account** Screenshot_2020-10-14 GiveWP Settings ‹ WordPress — WordPress(1) |
| **No connected Stripe accounts** Screenshot_2020-10-14 GiveWP Settings ‹ WordPress — WordPress |

## Pre-review Checklist

-   [x] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@since` tags included in DocBlocks
-   [x] Changelog updated
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

## Testing Instructions

1. Go to Settings -> Payment Gateways -> Stripe -> Manage Accounts.
1. Click "Connect with Stripe" and complete the connect process.
1. Go to Settings -> [Authorized Applications](https://dashboard.stripe.com/account/applications) in the Stripe dashboard of the connected account.
1. View GiveWP as an Authorized Application.
1. Go to Settings -> Payment Gateways -> Stripe -> Manage Accounts.
1. Click "Disconnect" to remove a connected account.
1. Go to Settings -> [Authorized Applications](https://dashboard.stripe.com/account/applications) in the Stripe dashboard of the connected account.
1. View GiveWP is still an Authorized Application.

### CODE REVIEW

**Comment by kjohnson** [comment]
I'm having issues building the `give-stripe` add-on locally, so I haven't tested that part to confirm.

Otherwise, this is ready for review.

**Comment by kevinwhoffman** [comment]
I'm going to handle review of this one following this morning's call.

**Comment by kevinwhoffman** [comment]
i discussed this briefly with @DevinWalker today. He is going to test locally and provide feedback on the UI.

File: includes/gateways/stripe/includes/admin/class-give-stripe-admin-settings.php [hunk 1]
```diff
@@ -759,25 +761,20 @@ class="stripe-btn-disabled give-hidden"
...
-
+									$disconnect_message = esc_html__( 'Are you sure you want to disconnect this Stripe account?', 'give' );
+									$disconnect_url     = add_query_arg(
+										[
+											'post_type'   => 'give_forms',
+											'page'        => 'give-settings',
+											'tab'         => 'gateways',
+											'section'     => 'stripe-settings',
+											'give_action' => ( 'connect' === $details['type'] )
+												? 'disconnect_connected_stripe_accout'
```
Comment (ravinderk) [inline includes/gateways/stripe/includes/admin/class-give-stripe-admin-settings.php:772]: @kjohnson Typo in string. Check suffix.

File: includes/gateways/stripe/includes/admin/class-give-stripe-admin-settings.php [hunk 2]
```diff
@@ -1063,6 +1073,19 @@ class="<?php echo esc_attr( $field_options['class'] ); ?>"
 			</tr>
 			<?php
 		}
+
+		/**
+		 * @since 2.8.0
+		 */
+		public function disconnect_connected_stripe_accout() {
+			$data = give_clean( $_GET );
```
Comment (ravinderk) [inline includes/gateways/stripe/includes/admin/class-give-stripe-admin-settings.php:1081]: @kjohnson Can we tighten security here:
- Add nonce validation
- Check for user permission (must have)

**Review by ravinderk — CHANGES_REQUESTED** [review]

File: includes/gateways/stripe/includes/admin/class-give-stripe-admin-settings.php (hunk 1 above)
Comment (kjohnson) [reply includes/gateways/stripe/includes/admin/class-give-stripe-admin-settings.php]: Good catch.

Fixed by commit 9b88f1c2bcf5c17e0d2974768f5da81f5b57815d

File: includes/gateways/stripe/includes/admin/class-give-stripe-admin-settings.php (hunk 2 above)
Comment (kjohnson) [reply includes/gateways/stripe/includes/admin/class-give-stripe-admin-settings.php:1084]: User permission check resolved by commit 9d10699caa4c7ab0bba3a34ec208889377d8b930

Comment (kjohnson) [reply includes/gateways/stripe/includes/admin/class-give-stripe-admin-settings.php:1084]: Nonce added with commit a812810353d0acce487aeaf65680ca43fe67131a

Comment (ravinderk) [reply includes/gateways/stripe/includes/admin/class-give-stripe-admin-settings.php:1084]: Looks good.

**Review by Benunc — APPROVED** [review]
This is ready to ship. I just tested it and it is working as expected. I'd like to see this released soon, but given that there's a merge conflict here and it's technically introducing new functionality, I'd feel better about this being in 2.10 and shipping after Thanksgiving and a bit of testing.

**Review by JasonTheAdams — APPROVED** [review]
This makes sense and is working for me! Go ahead and merge this.

**Pull request merged**